backend/src/main/scala/bloop/CompilerCache.scala

```diff
@@ -265,7 +281,13 @@ final class CompilerCache(
         processJavaDirArgument(cleanedOptions.indexOf("-s"))
         processJavaDirArgument(cleanedOptions.indexOf("-h"))
 
-        val newJavacOptions = cleanedOptions.toList.asJava
+        output.getSingleOutputAsPath match {
+          case p if p.isPresent => java.nio.file.Files.createDirectories(p.get)
+          case _ =>
+        }
+
+        val outputOption = CompilerArguments.outputOption(output)
```

Comment:
It's needed to specify the output directory now, otherwise it will get compiled next to the source file. The change is based on changes to zinc/sbt